keras/engine/training.py

```diff
@@ -1001,32 +1014,37 @@ def _fit_loop(self, f, ins, out_labels=None, batch_size=32,
                  `f` and the list of display names of the outputs of `f_val`.
             initial_epoch: epoch at which to start training
                 (useful for resuming a previous training run)
+            steps_per_epoch: Total number of steps (batches of samples)
+                before declaring one epoch finished and starting the
+                next epoch. Ignored with the default value of `None`.
 
         # Returns
             `History` object.
         """
         do_validation = False
         if val_f and val_ins:
             do_validation = True
-            if verbose:
+            if steps_per_epoch and not validation_steps:
+                raise ValueError('When using `steps_per_epoch` and validation data, '
+                                 'you must specify a value for `validation_steps`.')
```

Comment:
Once a graph is defined with input tensors it exists as a fixed network with that form and will draw all data from the input tensors any time it is run. Unfortunately, I'm not aware of a way to feed a validation numpy array into a Model instantiated with input tensors, and I think even validation tensors would require graph rewiring so I propose we say both possibilities remain out of scope for this PR.

What do you think of simply adding a check for any validation data when `fit()` is called with either x or y set to `None`, and providing an error message indicating validation on networks with input tensors are not yet supported?